tsl/src/continuous_aggs/materialize.c

```diff
@@ -459,7 +468,35 @@ get_materialization_end_point_for_table(int32 raw_hypertable_id, int32 materiali
 		*materializing_new_range = false;
 		return old_completed_threshold;
 	}
-
+	if (max_time < end_time)
+	{
+		int64 maxarg = max_time;
```

Comment:
I really struggle to follow the code in this function and keeping track of and knowing the difference between `max_time`, `end_time`, `min_time`, `start_time`, `now_time`, `maxarg`, etc.

I think this function requires a thorough explanation of what is going on here. Maybe even some ASCII illustrations with a timeline putting the above timestamps in relation to each other.

Reply:
this code  snippet has been simplified.